src/esp/assets/ResourceManager.cpp

```diff
@@ -1648,43 +1751,257 @@ void ResourceManager::loadMaterials(Importer& importer,
   }
 }  // ResourceManager::loadMaterials
 
+Cr::Containers::Optional<Mn::Trade::MaterialData>
+ResourceManager::createUniversalMaterial(
+    const Cr::Containers::Optional<Mn::Trade::MaterialData>& origMaterialData) {
+  // create a material, based on the passed material, that will have reasonable
+  // attributes to support any possible shader type. should only be called if
+  // we are not using the Material's natively specified shaderType
+
+  // NOLINTNEXTLINE(google-build-using-namespace)
+  using namespace Mn::Math::Literals;
+
+  // get source attributes from original material
+  Cr::Containers::Array<Mn::Trade::MaterialAttributeData> newAttributes;
+  arrayAppend(newAttributes, origMaterialData->attributeData());
+
+  const auto origMatTypes = origMaterialData->types();
+  // add appropriate attributes based on what is missing
+  // flat material already recognizes Phong and pbr, so don't have to do
+  // anything
+
+  // multiplicative magic number for scaling from PBR to phong,
+  // hacky method to attempt to balance Phong and PBR light intensity reactions
+  const float magicPhongScaling = 1.5f;
+
+  // whether the MnMatAttr::TextureMatrix has been set already
+  bool setTexMatrix = false;
+  if (~(origMatTypes & Mn::Trade::MaterialType::Phong)) {
+    // add appropriate values for expected attributes to support Phong from
+    // PbrMetallicRoughnessMaterialData
+
+    const auto& pbrMaterial =
+        origMaterialData->as<Mn::Trade::PbrMetallicRoughnessMaterialData>();
+
+    /////////////////
+    // calculate Phong values from PBR material values
+    ////////////////
+
+    // derive ambient color from pbr baseColor
+    const Mn::Color4 ambientColor = pbrMaterial.baseColor();
+
+    // If there's a roughness texture, we have no way to use it here. The safest
+    // fallback is to assume roughness == 1, thus producing no spec highlights.
+    // If pbrMaterial does not have a roughness value, it returns a 1 by
+    // default.
+    const float roughness =
+        pbrMaterial.hasRoughnessTexture() ? 1.0f : pbrMaterial.roughness();
+
+    // If there's a metalness texture, we have no way to use it here.
+    // The safest fallback is to assume non-metal.
+    const float metalness =
+        pbrMaterial.hasMetalnessTexture() ? 0.0f : pbrMaterial.metalness();
+
+    // Heuristic to map roughness to spec power.
+    // Higher exponent makes the spec highlight larger (lower power)
+    // example calc :
+    // https://www.wolframalpha.com/input/?i=1.1+%2B+%281+-+x%29%5E4.5+*+180.0+for+x+from+0+to+1
+    // lower power for metal
+    const float maxShininess = Mn::Math::lerp(250.0f, 120.0f, metalness);
+    const float shininess = 1.1f + powf(1.0f - roughness, 4.5) * maxShininess;
```

Comment:
I know this code is mostly a copypaste of the original, but while we're at it, let's use the guaranteed-to-also-have-a-float-overload API instead:

```suggestion
    const float shininess = 1.1f + Mn::Math::pow(1.0f - roughness, 4.5f) * maxShininess;
```

Same in other cases where `powf` is used, note the `f` literal used for the power factor.

Reply:
tried to get all the literals to be floats, guess I missed a few.  thanks! :)